src/Symfony/Bridge/Twig/Extension/AssetMapperRuntime.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Extension;
+
+use Symfony\Component\AssetMapper\AssetMapperInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+final class AssetMapperRuntime
+{
+    public function __construct(
+        private readonly AssetMapperInterface $assetMapper,
+    ) {
+    }
+
+    public function source(string $logicalPath): string
+    {
+        if (!$asset = $this->assetMapper->getAsset($logicalPath)) {
+            throw new \LogicException(\sprintf('The asset "%s" was not found.', $logicalPath));
+        }
+
+        if ($asset->content) {
+            return $asset->content;
+        }
+
+        if (false === $content = file_get_contents($asset->sourcePath)) {
```

Comment:
```suggestion
        if (false === $content = (new Filesystem())->readFile($asset->sourcePath)) {
```
?

As Filesystem is in AssetMapper dependencies..  and has now replaced all file_get_contents (afaik) ?

Reply:
Had this originally but `Filesystem::readFile()` is 7.1+. Wasn't sure if we should bump deps.